File: sdk/src/main/java/com/kin/ecosystem/history/view/IOrderHistoryView.java
package com.kin.ecosystem.history.view;

import android.support.annotation.NonNull;
import com.kin.ecosystem.base.IBaseView;
import com.kin.ecosystem.core.network.model.Order;
import com.kin.ecosystem.history.presenter.ICouponDialogPresenter;
import com.kin.ecosystem.history.presenter.IOrderHistoryPresenter;
import java.util.List;

public interface IOrderHistoryView extends IBaseView<IOrderHistoryPresenter>{

    void updateOrderHistoryList(List<Order> orders);

    void onItemInserted();

    void onItemUpdated(int index);

    void showCouponDialog(@NonNull final ICouponDialogPresenter presenter);
}

File: sdk/src/main/java/com/kin/ecosystem/history/view/IOrderHistoryView.kt
package com.kin.ecosystem.history.view

import com.kin.ecosystem.base.IBaseView
import com.kin.ecosystem.core.network.model.Order
import com.kin.ecosystem.history.presenter.OrderHistoryPresenter
import com.kin.ecosystem.main.INavigator

interface IOrderHistoryView : IBaseView {

    fun showEarnList()

    fun showSpendList()

    fun setEarnList(earnList: List<Order>)

    fun setSpendList(spendList: List<Order>)

    fun onEarnItemInserted()

    fun onSpendItemInserted()

    fun onEarnItemUpdated(index: Int)

    fun onSpendItemUpdated(index: Int)

    fun notifyEarnDataChanged(range: IntRange)

    fun notifySpendDataChanged(range: IntRange)

    fun setNavigator(navigator: INavigator)

    fun showMenuTouchIndicator(visibility: Boolean)

    fun updateSubTitle(amount: Int, orderStatus: OrderHistoryPresenter.OrderStatus, orderType: OrderHistoryPresenter.OrderType)
}
